content/300-guides/500-other/900-advanced-database-tasks/06-typesafe-raw-sql/01-safeql-gis.mdx

```diff
@@ -0,0 +1,128 @@
+---
+title: 'Using SafeQL with unsupported GIS features'
+metaTitle: 'Using SafeQL with unsupported GIS features (Guide)'
+metaDescription: 'Learn how to use SafeQL and Prisma Client extensions to work around features not natively supported by Prisma.'
+---
+
+## Overview
+
+This page explains how to use Prisma Client extensions and [SafeQL](https://safeql.dev) in order to make the experience of working with geographic data better.
+
+Our example will be using [PostGIS](https://postgis.net/) and PostgreSQL, but is applicable to other features and database engines.
+
+## Prerequisites
+
+To follow this guide, we recommend:
+
+- an existing [PostgreSQL](https://www.postgresql.org/) database
+- an existing Prisma project
+
+## What is SafeQL?
+
+[SafeQL](https://safeql.dev/) is an ESLint plugin that allows for advanced linting and type safety within raw SQL queries. After setup, SafeQL works with Prisma `$queryRaw` and `$executeRaw` to provide type safety when raw queries are required.
+
+## Geographic data support in Prisma
+
+At this time geographic data, and specifically PostGIS, are not supported by Prisma. Models that have geographic data columns will not have the benefit of type safety, nor will be able to use mutating operations like `create` and `update`.
+
+Prisma does support operations using `$queryRaw` and `$executeRaw` and we will be using these in order to create a Prisma Client extension with SafeQL to improve the usage of geographic data in Prisma.
+
+## 1. Set up Prisma for use with PostGIS
+
+If you haven't already, enable the `postgresqlExtensions` Preview feature and add the `postgis` PostgreSQL extension in your Prisma schema:
+
+```prisma highlight=3,8;add
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["postgresqlExtensions"]
+}
+
+datasource db {
+  provider   = "postgresql"
+  url        = env("DATABASE_URL")
+  extensions = [postgis]
+}
+```
+
+Then, create a migration using `npx prisma migrate dev` in order to enable the extension. The migration file should look like the following:
+
+```sql
+
+-- CreateExtension
+CREATE EXTENSION IF NOT EXISTS "postgis";
+
+```
+
+Make sure that the migration is applied!
```

Comment:
I think it would be important to state how for the reader 

```suggestion
Make sure that the migration is applied by running `prisma migrate status`.
```